### ISSUE TITLE
Latex Error when compiling including the \input{figure.tex}

### ISSUE THREAD

**Issue by Vnadon** [matlab2tikz/matlab2tikz#938]
Hi,

When I try to compile my Latex document including the matlab2tikz converted .tex figure file I get the following error:

! Missing } inserted.
<inserted text>
}
l.957 \end{axis}
I've inserted something that you may have forgotten.

Please find attached a minimal .tex document file with the tikz .tex file causing problem.
[example2.zip](https://github.com/matlab2tikz/matlab2tikz/files/383963/example2.zip)

**Comment by miscco** [comment]
@Vnadon Thank you for the issue. Can you please give some additional infomration:

1. A minimal working example of matlab code reproducing the issue
2. Which version if matlab/octave do you use.
3. What is the pfgplots version installed in your system
4. Did you try the latest development version?

**Comment by miscco** [comment]
@Vnadon Fro a quick glance at you file this seems to be a know issue regarding legends of special plots. Basically if matlab stores the legends in a different axis e.g. it does so for bode plots, then m2t adds the "forget plot" option to the respective addplot commands.

Now if there is still a legend around pgfplots cannot assign t and bails. So a simple short time solution would be to search for forget plot and get rid of it.

However, it seems that you have some complicated layers in your plot that mess with the labels.

Long story short we would need the matlab mwe in any case, but i would not get my hopes high

**Comment by egeerardyn** [comment]
I agree with @miscco: this is due to the legends. It compiles fine when I remove all `\addlegendentry` commands from your example.

Compiled, it then looks like:
<img width="571" alt="screen shot 2016-07-26 at 16 32 00" src="https://cloud.githubusercontent.com/assets/177360/17141959/8724c05a-534e-11e6-892f-a1ec59734b65.png">

so it seems to me that you are using a very customized routine to make box plots which you can adapt to avoid problems.

What is happening, is that you are attaching legends to plots consisting only of invisible points. Apparently, `pgfplots` doesn't like that too much when `unbounded coords=jump` is given. This is something that we have observed before, but it is something we should also report upstream to the pgfplots project.

There is an easy **work-around** for this: attach the legends to points that you place outside of the visible axis (`xlim` and `ylim` are your friends there).

**Comment by Vnadon** [comment]
Tikz version is 2013/12/13 v3.0.0
pgfplots version 2014/08/04 v1.11

Matlab2tikz version 1.0.0

Here is an example of the Matlab script I'm using (Matlab version 2012B x64)


[example_boxplot_tikz.zip](https://github.com/matlab2tikz/matlab2tikz/files/384071/example_boxplot_tikz.zip)

Should I put the legend outside the plot, like using NorthEastOutside for example?

**Comment by miscco** [comment]
@Vnadon as @egeerardyn said the best thing would be to just add points outside of the visible axis. It seems that NaN does not do the trick and neither does inf. 

However be aware, that you should run cleanfigure beforehand otherwise it would remove those points.

**Comment by Vnadon** [comment]
cleanfigure removes the x-axis, at this point I'd rather remove the legend completely and add it in the .tex file of the figure can I do that? how do I do it?
